numpy/lib/tests/test_packbits.py

```diff
@@ -17,6 +17,22 @@ def test_packbits():
     assert_raises(TypeError, np.packbits, np.array(a, dtype=float))
 
 
+def test_packbits_empty():
+    for dtype in [np.bool, np.uint8, np.int]:
```

Comment:
Why just these three types? And why just the last dimension zero? Tests should be exhaustive. Admittedly, most aren't but, strangely enough, when improved often reveal errors.

All integer types are available in `np.typecodes['AllInteger']`,  the boolean type is separate. Or you could just use the string `'?bBhHiIlLqQ'` for typecodes. If boolean is acceptable I think the docstring need improvement also.